### PR TITLE
Add EA extensibility point to allow go-to-def to navigate to F# source symbols.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Note: this will not work for 'peek definition' currently due to that system needing complex information about the actual file/span to present. 

if peek is important, we'll need to come up with some additional approaches to make that work.

### CODE REVIEW

File: src/Tools/ExternalAccess/FSharp/Navigation/IFSharpCrossLanguageSymbolNavigationService.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using Microsoft.CodeAnalysis.Navigation;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Navigation
+{
+    /// <inheritdoc cref="ICrossLanguageSymbolNavigationService"/>
+    internal interface IFSharpCrossLanguageSymbolNavigationService
+    {
+        /// <inheritdoc cref="ICrossLanguageSymbolNavigationService.TryGetNavigableLocationAsync"/>
+        Task<IFSharpNavigableLocation?> TryGetNavigableLocationAsync(string documentationCommentId, CancellationToken cancellationToken);
+    }
```
Comment (CyrusNajmabadi) [inline src/Tools/ExternalAccess/FSharp/Navigation/IFSharpCrossLanguageSymbolNavigationService.cs:16]: @vzarytovskii this is the api you would implement and export.  LMK if the contract of it makes sense to you and is feasible for F# to implement.

File: src/Features/Core/Portable/Navigation/ICrossLanguageSymbolNavigationService.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    /// a language like F# to navigate to its source definition of a symbol that roslyn views as a metadata symbol.
+    /// </summary>
+    internal interface ICrossLanguageSymbolNavigationService
+    {
+        /// <summary>
+        /// Attempts to get location that can be navigated to for a particular symbol id.  The symbol id format is
+        /// defined at: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/#id-strings.  Should
+        /// return <see langword="null"/> if the 3rd party language cannot navigate to this particular symbol.
+        /// </summary>
+        Task<INavigableLocation?> TryGetNavigableLocationAsync(string documentationCommentId, CancellationToken cancellationToken);
```
Comment (ryzngard) [inline src/Features/Core/Portable/Navigation/ICrossLanguageSymbolNavigationService.cs:21]: nit: could we name this `xmlDocumentationCommentId`? Until I read the doc I wasn't actually sure what this had to do with getting a location.

**Review by ryzngard — APPROVED** [review]
LGTM

File: src/Features/Core/Portable/Navigation/ICrossLanguageSymbolNavigationService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Navigation/ICrossLanguageSymbolNavigationService.cs:21]: gotcha.  i think i'm ok with the name given that our own public apis are named `GetDocumentationCommentId()`.

**Pull request merged**